prereise/gather/demanddata/uci/HiGrid_Elec_Trans4_8_nths.py

```diff
@@ -0,0 +1,195 @@
+    '''determines month of each day
+
+    :param list[int] month_days: a list of integers where each value in the list represents the number of days in that month (0 indexed)
+    :param list[int] month_first_day: a list of integers where each value in the list represents the day of the week of the first day of that month (0 indexed)
+    :return: np.ndarray -- stores the month the day is in for each day of the year
+
+    '''
+    # initalize day counter
+    D_count = 0
+
+    # initialize integers because all zeros
+    inputMonth = np.zeros(365, int)
+
+    # iterate over 12 months
+    for i in range(len(month_days)):
+
+        #initialize first day of each month for use in the next for loop
+        k = month_first_day[i]
+
+        #Iterate over the day of each month
+        for j in range(month_days[i]):
+            # Populate inputMonth with month of each day ex: inputMonth[100] gives us month number of day 101
+            inputMonth[D_count] = i
+
+            #increment day counter
+            D_count += 1
+
+    return inputMonth
+
+
+def get_month2(data: np.array(list[float or int]])) -> np.array(int):
+    '''gets month2 value from data
+    :param np.array(list[float or int]) data: the data to get months from
+    :return: np.array(int) -- list of months
+    '''
+    return np.array([x[5] % 100 for x in data], int)
+
+
+def get_dayofweek2(data: np.array(list[float or int]])) -> np.array(int):
+    '''gets dayofweek2 value from data
+    :param np.array(list[float or int]) data: the data to get day of week from
+    :return: np.array(int) -- the day of the week for each entry in data
+    '''
+    return np.array([x[6] for x in data], int)
+
+def get_day2(data: np.array(list[float or int]])) -> np.array(int):
+    '''gets day2 value from data
+    :param np.array(list[float or int]) data: the data to get day of week from
+    :return: np.array(int) -- indicates weekend or weekday for every day
+    '''
+    return np.array([x[7] for x in data])
+
+def load_data(census_1: int) -> list[list[float or int]]:
+    '''loads the data at nhts_census.mat
+    :param int census_1: the type of census
+    :return: list[list] -- the data loaded from nths_census.mat
+    '''
+    # somehow load in 'nhts_census.mat'
+    nhts_census = scipy.io.loadmat('nhts_census.mat')
+
+    # census_1 will be 1-9 incluse, so i in range(1,10)
+    for i in range(1,10):
+        # return the proper one
+        if census_1 == i:
+            return nhts_census[f'census_{i}_sorted']
+
+
+def remove_ldt(newdata: list[list[float or int]]) -> list[list[float or int]]:
+    '''removes light duty trucks from data loaded from nths_census.mat
+    :param list[list] newdata: the data returned from load_data
+    :return: list[list] -- the data loaded from load_data with lal rows involving LDT removed
+    '''
+    # keep track of which rows to delete
+    rows_to_delete = set()
+    for vtype in range(len(newdata)):
+        # >4 for LDV, <5 for LDT

```

Comment:
This comment (and the associated code operation) seems to be opposite from the description in the document in the dropbox, which says: `Vehicle type (1-4 LDV, 5+ LDT)`.